pyiceberg/io/__init__.py

```diff
@@ -320,6 +331,13 @@ def _infer_file_io_from_scheme(path: str, properties: Properties) -> Optional[Fi
     return None
 
 
+def _get_first_property_value(properties: Properties, property_names: Tuple[str, ...]) -> Optional[Any]:
```

Comment:
Nit: Duplicate function defined here and in pyiceberg/catalog/__init__.py could be consolidated to one

Reply:
I find this PR very hard to read with all the constants. I would like to do a suggestion, how about changing the signature to:
```suggestion
def _get_first_property_value(properties: Properties, *property_names: str) -> Optional[Any]:
```

And avoid the additional constant with the grouped values:

```python
_get_first_property_value(self.properties, S3_ACCESS_KEY_ID_PROPERTIES)
```

Instead, write:

```python
_get_first_property_value(self.properties, S3_ACCESS_KEY_ID, AWS_ACCESS_KEY_ID)
```

The `S3_ACCESS_KEY_ID_PROPERTIES` seemed to be used just once. WDYT?

Reply:
Thanks for these great suggestions! I've taken both by moving the method to `PropertyUtil` and changing the signature as suggested